core/api/src/main/java/org/eclipse/sensinact/core/notification/LifecycleNotification.java

```diff
@@ -23,13 +23,9 @@
  *
  * LIFECYCLE/&lt;model&gt;/&lt;provider&gt;[/&lt;service&gt;[/&lt;resource&gt;]]
  */
-public class LifecycleNotification extends AbstractResourceNotification {
-
-    public Status status;
-
-    public Object initialValue;
-
-    public Map<String, Object> initialMetadata;
+public record LifecycleNotification(String modelPackageUri, String model, String provider,
```

Comment:
may encapsulate this fields in en extra record, that is then used in all records

String modelPackageUri, String model, String provider,
        String service, String resource,

Reply:
> may encapsulate this fields in en extra record

Do you mean a default visibility record? I wouldn't want to make it a public type as it's not something that really forms part of the public API, and we don't want people to expect that they'll get an instance. Making it default visibility would then mean all public records implementing methods which delegate to the default visibility type. The interface we have already guarantees that we don't misspell the names of the record components, so I don't really see the value here.